feat: Add split button component

JKMarkowski opened the pull request:
#### Please provide a link to the associated issue.
fixes https://github.com/SAP/fundamental-ngx/issues/949
fixes https://github.com/SAP/fundamental-ngx/issues/652
#### Please provide a brief summary of this pull request.
I added split button component.
#### If this is a new feature, have you updated the documentation?
Yes there is completely new page with documentation about split button and some examples.

droshev commented:
@JKMarkowski could you resolve the conflicts?

JKMarkowski commented:
Yes, conflicts resolved.

droshev commented:
@MattL75 @mikerodonnell89

MattL75 commented on docs/app/documentation/component-docs/button-split/button-split-header/button-split-header.component.html:
```diff
@@ -0,0 +1,11 @@
+<header>Button</header>
```
Change to "Split Button", this also applies to every instance where "Button Split" is used. Fundamental-Styles calls it Split Button so we should follow their lead on this.

MattL75 commented on docs/app/documentation/component-docs/button-split/examples/button-split-examples.component.ts:
```diff
@@ -0,0 +1,37 @@
+})
+export class ButtonSplitProgrammaticalExampleComponent {
+    isOpen: boolean = false;
+}
```
add a empty line between the components

MattL75 commented on docs/app/documentation/component-docs/button-split/examples/button-split-template-example.component.html:
```diff
@@ -0,0 +1,17 @@
+<fd-button-split>
+    <ng-template fd-button-split-action-title>
+        <u>Action Underscored</u>
```
Better to make it more obvious that this is for complex action titles. Maybe adding an icon on the left side of the text instead?

MattL75 commented on library/src/lib/button-split/button-split-utils/button-split.directives.ts:
````diff
@@ -0,0 +1,87 @@
+ *</fd-button-split>
+ *```
+ */
+
````
remove empty line between the comment and directive annotation

MattL75 commented on library/src/lib/button-split/button-split.component.ts:
```diff
@@ -0,0 +1,128 @@
+    selector: 'fd-button-split',
+    templateUrl: 'button-split.component.html'
+})
+export class ButtonSplitComponent extends AbstractFdNgxClass {
```
If it's not needed, better to not extend AbstractFdNgxClass. I'm currently looking to progressively phase this out as we go

mikerodonnell89 commented:
I think the width of the popover body should match the width of the popover control

JKMarkowski commented:
@MattL75  I changed the component naming which unfortunately caused all of yours comments outdated. About the accessibility, I went through it with screen reader and it worked same as it is on our current popover/dropdown.

@mikerodonnell89 I added default popover body matching width of popover control. It can be disabled anytime by user.

JKMarkowski commented:
Sure I removed links and added some alerts on click. I also think that it's not great for user, when he is redirected to home page after clicking on link. We should change it on other components.

MattL75 commented:
@JKMarkowski Yeah I agree, thanks for changing it

MattL75 commented on docs/app/documentation/component-docs/split-button/split-button-docs.component.html:
```diff
@@ -0,0 +1,55 @@
+<separator></separator>
+<h2>Open/Close programmatically</h2>
+<description>
+    There is possibility to control open/close of popover by 2-ways data binding <code>[(isOpen)]</code>
```
It is possible to control the open state of the popover through two-way binding to the isOpen property.

MattL75 commented on docs/app/documentation/component-docs/split-button/split-button-docs.component.html:
```diff
@@ -0,0 +1,55 @@
+<code-example [code]="buttonSplitIcons" [language]="'HTML'"></code-example>
+
+<separator></separator>
+<h2>Open/Close programmatically</h2>
```
Open/Close Programatically

(just missing capital P in programatically)

MattL75 commented on docs/app/documentation/component-docs/split-button/split-button-docs.component.html:
```diff
@@ -0,0 +1,55 @@
+<code-example [code]="buttonSplitProgrammaticallyExample" [language]="'HTML'"></code-example>
+
+<separator></separator>
+<h2>Adding template to action button</h2>
```
Custom Template for Action Button

MattL75 commented on docs/app/documentation/component-docs/split-button/split-button-docs.component.html:
```diff
@@ -0,0 +1,55 @@
+<separator></separator>
+<h2>Adding template to action button</h2>
+<description>
+    There is possibility to add some html template to action button with <code>fd-split-button-action-title</code> directive
```
It is possible to add a custom template to the action button through the <code>fd-split-button-action-title</code> directive.

MattL75 commented on docs/app/documentation/component-docs/split-button/split-button-header/split-button-header.component.html:
```diff
@@ -0,0 +1,11 @@
+<header>Split Button</header>
+<description>
+    The Split Button component is made by connecting popover component and button directive.
```
The split button component is composed of the popover and button components.

JKMarkowski commented:
All of these changed are resolved now. Thanks for checking it, I really should focus more on documentation, instead of rushing.

mikerodonnell89 commented:
It does need keyboard support for selecting options.  Refer to the combobox component keydown handlers for examples on how this should work

JKMarkowski commented:
Hi @mikerodonnell89 , @MattL75 I am a little confused, because for dropdown/popover/default menu component we do not support arrowup/arrowdown navigation, I was more thinking about implementing some keyboard handler functionality on menu component. A lot of components are based on this menu component and it would provide functionallity to all of them. What do you think about it ?
